cluster.go

```diff
@@ -1135,8 +1124,9 @@ func (c *Cluster) allocate(hash *cid.Cid, repl int) ([]peer.ID, error) {
 
 		// we don't have enough peers to pin
 		if len(candidateAllocs) < needed {
```

Comment:
If `candidateAllocs` is just `candidates` sorted by priority, can't we check `len(candidates) < needed` before we call `Allocate`? Would save a sorting step if we don't have enough candidates anyway.